### PR TITLE
Add "Hard Rock CoffeeScript"

### PR DESCRIPTION
**Pull request by chauxvive**
Added "Hard Rock CoffeeScript" and link

## What does this PR do?
Add resource(s) | Remove resource(s) | Add info | Improve repo

### Description
This is a full book published under the "MIT license", or so the internet tells me.

### Why is this valuable (or not)?
This seems like a nice breakdown of CoffeeScript for beginners.

### How do we know it's really free?
This whole book is here in the author's git.

### For book lists, is it a book? For course lists, is it a course? etc.
It is indeed a book!

## Checklist:
- [x] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/master/CONTRIBUTING.md)
- [x] Search for duplicates.
- [x] Include author(s) and platform where appropriate.
- [x] Put lists in alphabetical order, correct spacing.
- [x] Add needed indications (PDF, access notes, under construction)

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

File: books/free-programming-books-langs.md [hunk 1]
```diff
@@ -569,6 +569,7 @@ That section got so big, we decided to split it into its own file, the [BY SUBJE
 
 * [CoffeeScript Cookbook](https://coffeescript-cookbook.github.io)
 * [CoffeeScript Ristretto](https://leanpub.com/coffeescript-ristretto/read) - Reginald Braithwaite  *(Leanpub account or valid email requested)*
+* [Hard Rock CoffeeScript](https://alchaplinsky.github.io/hard-rock-coffeescript/)
 * [Smooth CoffeeScript](http://autotelicum.github.io/Smooth-CoffeeScript/SmoothCoffeeScript.html)
 * [The Little Book on CoffeeScript](http://arcturo.github.io/library/coffeescript/)
```
Comment (davorpa) [inline books/free-programming-books-langs.md:574]:
```suggestion
* [The Little Book on CoffeeScript](http://arcturo.github.io/library/coffeescript/) - Alex Chaplinsky (gitbook)
```

**Review by davorpa — CHANGES_REQUESTED** [review]

**Comment by davorpa** [comment]
It's nice see welcome back!! #2480

File: books/free-programming-books-langs.md [hunk 2]
```diff
@@ -569,8 +569,9 @@ That section got so big, we decided to split it into its own file, the [BY SUBJE
 
 * [CoffeeScript Cookbook](https://coffeescript-cookbook.github.io)
 * [CoffeeScript Ristretto](https://leanpub.com/coffeescript-ristretto/read) - Reginald Braithwaite  *(Leanpub account or valid email requested)*
+* [Hard Rock CoffeeScript](https://alchaplinsky.github.io/hard-rock-coffeescript/)
 * [Smooth CoffeeScript](http://autotelicum.github.io/Smooth-CoffeeScript/SmoothCoffeeScript.html)
-* [The Little Book on CoffeeScript](http://arcturo.github.io/library/coffeescript/)
+* [The Little Book on CoffeeScript](http://arcturo.github.io/library/coffeescript/) - Alex Chaplinsky (gitbook)
```
Comment (davorpa) [inline books/free-programming-books-langs.md:574]: Sorry for the mistake in latest review.

Revert or complete with real authors:

 ```suggestion
* [The Little Book on CoffeeScript](http://arcturo.github.io/library/coffeescript/) - Alex MacCaw, David Griffiths, Satoshi Murakami, Jeremy Ashkenas
```

**Review by davorpa — CHANGES_REQUESTED** [review]
Sorry for the mistake in latest review. https://github.com/EbookFoundation/free-programming-books/pull/6543#commitcomment-59012610

**Review by davorpa — APPROVED** [review]

**Comment by eshellman** [comment]
Thanks, and welcome to free-programming-books!

**Pull request merged**